File: app/src/main/java/com/github/pavlospt/microsoftassignment/di/components/AssignmentAppComponent.java
package com.github.pavlospt.microsoftassignment.di.components;

import com.github.pavlospt.microsoftassignment.beer_list_feature.BeerListActivity;
import com.github.pavlospt.microsoftassignment.di.modules.AssignmentAppModule;
import com.github.pavlospt.microsoftassignment.di.modules.AssignmentStorIOModule;
import dagger.Component;
import javax.inject.Singleton;

@Singleton
@Component(modules = {
    AssignmentAppModule.class, AssignmentStorIOModule.class
})
public interface AssignmentAppComponent {
  void inject(BeerListActivity beerListActivity);
}

File: app/src/main/java/com/github/pavlospt/microsoftassignment/di/components/AssignmentAppComponent.kt
package com.github.pavlospt.microsoftassignment.di.components

import com.github.pavlospt.microsoftassignment.beerListFeature.BeerListActivity
import com.github.pavlospt.microsoftassignment.di.modules.AssignmentAppModule
import com.github.pavlospt.microsoftassignment.di.modules.AssignmentStorIOModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AssignmentAppModule::class, AssignmentStorIOModule::class))
interface AssignmentAppComponent {
  fun inject(beerListActivity: BeerListActivity)
}